protocol/dataunit/dataunit.go

```diff
@@ -0,0 +1,111 @@
+// Package dataunit implements low-level encoding of the EPP data unit protocol.
+//
+// The data unit protocol is a simple framing of an XML payload prefixed with a 4-byte header
+// in network byte order that expresses the total length of the data unit (header + payload size).
+// EPP data units are sent and received via an underlying transport (typically a TLS connection).
+//
+// See [RFC 5730] (Extensible Provisioning Protocol) and
+// [RFC 5734] (EPP Transport over TCP) for more information.
+//
+// [RFC 5730]: https://datatracker.ietf.org/doc/rfc5730/
+// [RFC 5734]: https://datatracker.ietf.org/doc/rfc5734/
+package dataunit
+
+import (
+	"context"
+	"encoding/binary"
+	"io"
+)
+
+// Receive receives a single EPP data unit from [io.Reader] r, returning the payload or an error.
+// If the EPP data unit read has zero-length, Receive will return (nil, nil).
```

Comment:
Why not some sort of error like `io.UnexpectedEOF`? I'm sure there's a reason, perhaps worth adding to the doc.